ngraph/test/util/test_environment.cpp

```diff
@@ -0,0 +1,165 @@
+// Copyright (C) 2019-2020 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+#include <fstream>
+#include <pugixml.hpp>
+
+#include "test_environment.hpp"
+
+namespace ngraph
+{
+    namespace test
+    {
+        Summary* Summary::p_instance = nullptr;
+        SummaryDestroyer Summary::destroyer;
+
+        SummaryDestroyer::~SummaryDestroyer() { delete p_instance; }
+        void SummaryDestroyer::initialize(Summary* p) { p_instance = p; }
+        Summary& Summary::getInstance()
+        {
+            if (!p_instance)
+            {
+                p_instance = new Summary();
+                destroyer.initialize(p_instance);
+            }
+            return *p_instance;
+        }
+
+        void Summary::updateOPsStats(const std::shared_ptr<ngraph::Function>& function,
+                                     PassRate::Statuses status)
+        {
+            if (!function)
+            {
+                return;
+            }
+            for (const auto& op : function->get_ordered_ops())
+            {
+                updateOPsStats(op, status);
+            }
+        }
+        void Summary::updateOPsStats(const std::shared_ptr<ngraph::Node>& op,
+                                     PassRate::Statuses status)
+        {
+            if (!op || ngraph::is_type<ngraph::op::Parameter>(op) ||
+                ngraph::is_type<ngraph::op::Constant>(op) ||
+                ngraph::is_type<ngraph::op::Result>(op))
+            {
+                return;
+            }
+            else if (ngraph::is_type<ngraph::op::TensorIterator>(op))
+            {
+                updateOPsStats(op->get_type_info(), status);
+                auto ti = ngraph::as_type_ptr<ngraph::op::TensorIterator>(op);
+                updateOPsStats(ti->get_function(), status);
+            }
+            else if (ngraph::is_type<ngraph::op::v5::Loop>(op))
+            {
+                updateOPsStats(op->get_type_info(), status);
+                auto loop = ngraph::as_type_ptr<ngraph::op::v5::Loop>(op);
+                updateOPsStats(loop->get_function(), status);
+            }
+            else
+            {
+                updateOPsStats(op->get_type_info(), status);
+            }
+        }
+
+        void Summary::updateOPsStats(ngraph::NodeTypeInfo op, PassRate::Statuses status)
+        {
+            // TODO: Do we need to count skips?
+            auto& passrate = opsStats[op];
+            switch (status)
+            {
+            case PassRate::PASSED: passrate.passed += 1; break;
+            case PassRate::FAILED: passrate.failed += 1; break;
+            case PassRate::SKIPPED: passrate.skipped += 1; break;
+            }
+        }
+
+        void TestEnvironment::TearDown() { writeReport(); }
+
+        void TestEnvironment::writeReport(const std::string& reportFileName) const
+        {
+            std::vector<ngraph::OpSet> opsets;
```

Comment:
Looks like code duplication. May be it would be better to link `InferenceEngine test utils`? I mean we will keep this code in one place

Reply:
The problem is that we can disable nGraph unit test if set the flag `NGRAPH_UNIT_TEST_ENABLE=OFF` (we also disable this flag if `ENABLE_TESTS` is `OFF`)
But if we are talking about `ie_test_utils`, this library exists if `ENABLE_TESTS` flag is `ON`, it means that at the current moment we cannot have source in ngraph tests because it can be disabled in some cases.

Also ie backend already depends on inference_engine.

Reply:
One naive solution which came to my mind is create yet another static library (somewhere in `openvino/test/run_report_test`)  build it when `ENABLE_TESTS` is `ON` and link it to ngraph and ie unit test.
Is this fine solution?